File: src/main/java/com/core/base/corebase/domain/review/ReviewPeriod.java
package com.core.base.corebase.domain.review;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Getter
@Builder
@Embeddable
@NoArgsConstructor
@AllArgsConstructor
public class ReviewPeriod {
    private LocalDate reviewStartDate;
    private LocalDate reviewEndDate;

    public boolean isBefore(LocalDate date) {
        return reviewStartDate.isBefore(date);
    }

    public boolean between(LocalDate date) {
        return reviewStartDate.isEqual(date) ||
               reviewEndDate.isEqual(date) ||
               (reviewStartDate.isBefore(date) && reviewEndDate.isAfter(date));
    }
}

File: src/main/kotlin/com/core/base/corebase/domain/review/ReviewPeriod.kt
package com.core.base.corebase.domain.review

import jakarta.persistence.Embeddable
import lombok.Builder
import java.time.LocalDate


@Builder
@Embeddable
class ReviewPeriod(
    val reviewStartDate: LocalDate,
    val reviewEndDate: LocalDate,
){

    fun isBefore(now: LocalDate) : Boolean = reviewStartDate.isBefore(now)

    fun between(now : LocalDate) : Boolean =
        reviewStartDate.isEqual(now)
                  || reviewEndDate.isEqual(now)
                  || (reviewStartDate.isAfter(now) && reviewEndDate.isBefore(now))

}